decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/data/BasesMask.java

```diff
@@ -0,0 +1,310 @@
+            return new BasesMask(m.group(1), null, m.group(2), m.group(3), m.group(4), m.group(5), m.group(6), null);
+        }
+
+        p = Pattern.compile("y(\\*|\\d+),i(\\*|\\d+),i(\\*|\\d+),y(\\*|\\d+)", Pattern.CASE_INSENSITIVE);
+        if (p.matcher(basesMaskString).matches()) {
+            Matcher m = p.matcher(basesMaskString);
+            m.find();
+            return new BasesMask(m.group(1), null, m.group(2), null, m.group(3), null, m.group(4), null);
+        }
+
+        p = Pattern.compile("y(\\*|\\d+),i(\\*|\\d+),n(\\*|\\d+),y(\\*|\\d+)", Pattern.CASE_INSENSITIVE);
+        if (p.matcher(basesMaskString).matches()) {
+            Matcher m = p.matcher(basesMaskString);
+            m.find();
+            return new BasesMask(m.group(1), null, m.group(2), null, null, m.group(3), m.group(4), null);
+        }
+
+        p = Pattern.compile("y(\\*|\\d+),n(\\*|\\d+),i(\\*|\\d+),y(\\*|\\d+)", Pattern.CASE_INSENSITIVE);
+        if (p.matcher(basesMaskString).matches()) {
+            Matcher m = p.matcher(basesMaskString);
+            m.find();
+            return new BasesMask(m.group(1), null, null, m.group(2), m.group(3), null, m.group(4), null);
+        }
+
+        //single barcode patterns
+        p = Pattern.compile("y(\\*|\\d+)n(\\*|\\d+),i(\\*|\\d+)n(\\*|\\d+),y(\\*|\\d+)n(\\*|\\d+)", Pattern.CASE_INSENSITIVE);
+        if (p.matcher(basesMaskString).matches()) {
+            Matcher m = p.matcher(basesMaskString);
+            m.find();
+            return new BasesMask(m.group(1), m.group(2), m.group(3), m.group(4), null, null, m.group(5), m.group(6));
+        }
+
+        p = Pattern.compile("y(\\*|\\d+),i(\\*|\\d+)n(\\*|\\d+),y(\\*|\\d+)", Pattern.CASE_INSENSITIVE);
+        if (p.matcher(basesMaskString).matches()) {
+            Matcher m = p.matcher(basesMaskString);
+            m.find();
+            return new BasesMask(m.group(1), null, m.group(2), m.group(3), null, null, m.group(4), null);
+        }
+
+        p = Pattern.compile("y(\\*|\\d+),i(\\*|\\d+),y(\\*|\\d+)", Pattern.CASE_INSENSITIVE);
+        if (p.matcher(basesMaskString).matches()) {
+            Matcher m = p.matcher(basesMaskString);
+            m.find();
+            return new BasesMask(m.group(1), null, m.group(2), null, null, null, m.group(3), null);
+        }
+
+        throw new IllegalArgumentException("Unsupported bases mask string [" + basesMaskString + "]");
+    }
+
+    public static class BasesMaskBuilder {
+
+        private Integer readOneIncludeLength = Integer.MAX_VALUE;
+        private Integer readOneIgnoreLength = null;
+        private Integer indexOneIncludeLength = Integer.MAX_VALUE;
+        private Integer indexOneIgnoreLength = Integer.MAX_VALUE;
+        private Integer indexTwoIncludeLength = null;
+        private Integer indexTwoIgnoreLength = null;
+        private Integer readTwoIncludeLength = Integer.MAX_VALUE;
+        private Integer readTwoIgnoreLength = null;
+
+        public BasesMaskBuilder() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)